.github/workflows/build_and_test.yml

```
@@ -102,13 +102,86 @@ jobs:
+          path: p4sde
+
+      - name: 'Install dependencies'
+        working-directory: ./p4sde/tools/setup
+        run: |
+              sudo apt update -y
+              sudo apt install python python3-setuptools python3-pip python3-wheel python3-cffi sudo libssl-dev git zip libatomic1 libnl-route-3-dev curl tree -y
+              pip3 install distro
+              python3 install_dep.py
+              cmake --version
+              sudo apt-get purge cmake
+              sudo rm -rf /usr/local/bin/cmake || true
+              sudo rm -rf ~/.local/bin/cmake || true
+              # removing cmake 3.25 and reinstalling 3.20, as build is failing with cmake 3.25
+              wget https://cmake.org/files/v3.20/cmake-3.20.2-linux-x86_64.sh -O ${GITHUB_WORKSPACE}/cmake-3.20.2-linux-x86_64.sh
+              chmod a+x ${GITHUB_WORKSPACE}/cmake-3.20.2-linux-x86_64.sh
+              sudo mkdir -p /opt/rh/cmake
+              cd ${GITHUB_WORKSPACE} && sudo sh cmake-3.20.2-linux-x86_64.sh --prefix=/opt/rh/cmake --skip-license
+              sudo ln -s /opt/rh/cmake/bin/cmake /usr/local/bin/cmake
+              cmake --version
+
+      - name: 'Checkout Networking Recipe'
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+          path: ipdk.recipe
+
+      - name: 'Compile P4 dpdk target'
+        run: |
+             export SDE=${GITHUB_WORKSPACE}
+             mkdir install deps
+             export SDE_INSTALL=${GITHUB_WORKSPACE}/install
+             cd ${GITHUB_WORKSPACE}/p4sde
+             ./autogen.sh
+             ./configure --prefix=$SDE_INSTALL
+             make -j6
+             make install
+
+      - name: 'Build,Configure and Execute infrap4d dependencies'
+        run: |
+             cmake --version
+             cd ${GITHUB_WORKSPACE}/ipdk.recipe
+             export IPDK_RECIPE=`pwd`
+             cd $IPDK_RECIPE/setup
+             cmake -B build -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/deps
+             cmake --build build -j6
+
+      - name: 'Run Krnlmon Unit Tests for DPDK Target'
+        run: |
+             cd ${GITHUB_WORKSPACE}/ipdk.recipe
+             export IPDK_RECIPE=${GITHUB_WORKSPACE}/ipdk.recipe
+             export SDE_INSTALL=${GITHUB_WORKSPACE}/install
+             export DEPEND_INSTALL=${GITHUB_WORKSPACE}/deps
+             cmake -B build -DDPDK_TARGET=on -DWITH_OVSP4RT=off
+             cmake --build build -j4 --target krnlmon-test || true
+             cat build/Testing/Temporary/LastTest.log
+             zip krnlmon_ut_logs.zip build/Testing/Temporary/LastTest.log
+
+      - name: 'Upload krnlmon-ut-result artifact'
+        uses: actions/upload-artifact@v3
+        with:
+          name: krnlmon_ut_logs
+          path: ipdk.recipe/krnlmon_ut_logs.zip
+
   py_bandit_test:
      timeout-minutes: 10
-     runs-on: ubuntu-20.04
+     runs-on: ubuntu-latest
      steps:
-       - name: 'checkout Networking Recipe'
+       - name: 'Checkout Networking Recipe'
```

```suggestion
       - name: 'Check Out Networking Recipe'
```
When used as a verb, this should be _check out_ (two words).
_checkout_ (one word) is a noun or adjective.

---

Also, Stratum tests failed with latest ubuntu, works only with ubuntu 20.04 - example: https://github.com/ipdk-io/networking-recipe/actions/runs/5527966086/jobs/10110937201